expression/reconstructor/reconstructor.go

```diff
@@ -155,3 +162,18 @@ func codePretty(bytes []byte) string {
 
 	return fmt.Sprintf("0x%s", encoded)
 }
+
+func bech32Pretty(bytes []byte) string {
```

Comment:
I think this should be merged with addressPretty.

The only hint we have is the bytes in the address. See line 134 in this file. There the address reconstructor will produce a hex address if all else fails. Bech32 reconstruction can be added there.

Reply:
I mean, keep the bech32Pretty method separate, but call it from addressPretty in the else case.